airflow/providers/amazon/aws/operators/bedrock.py

```diff
@@ -0,0 +1,118 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import json
+from functools import cached_property
+from typing import TYPE_CHECKING, Sequence
+
+from airflow.models import BaseOperator
+from airflow.providers.amazon.aws.hooks.bedrock import BedrockRuntimeHook
+from airflow.utils.helpers import prune_dict
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+DEFAULT_CONN_ID = "aws_default"
+
+
+class BedrockInvokeModelOperator(BaseOperator):
```

Comment:
Maybe better to subclass of [`AwsBaseOperator`](https://github.com/apache/airflow/blob/6029c71e2c4a8f6bb7086728f4bc09f43d18c11c/airflow/providers/amazon/aws/operators/base_aws.py#L32) and use other helpers. This will avoid to define common arguments and common properties (such as hook)

Reply:
I closed this because I realized that every model that bedrock supports has a different input and output format, unfortunately.  For example, if you are using the Amazon Titan model, you have to provide an "inputText", but if you are using the Meta Llama model you have to provide a "prompt".  And the return formats are different as well.

I'm going to resubmit it in a more generic format that will work with any of their models, but I'll make your change in the new version.

Reply:
> Meta Llama model you have to provide a "prompt"

And I guess it also required to be properly formatted 🙄 

BTW [`invoke_model`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/bedrock-runtime/client/invoke_model.html) method accept abstract `body` input, so maybe starts with that implementation and after that add some helpers, which might provide ability to build/check inputs?

Reply:
Yup, that's what I am doing.   Just waiting for static checks and build-docs to finish up and I'll get it up.   I had really wanted the operator to just take a prompt and a model name, and return the generated response.  But (for now?) it'll take a user-formatted body and a model name and return the full response.  Which is a shame.  But maybe Some Day:tm: we can do per-model operators, or add some helpers that look up the expected body format for the given model name and abstract that out, or maybe Bedrock will eventually start adding a wrapper that handles the various formatting issues.... but yeah, for now it'll be a "thin operator" I guess.  I've added more to the documentation so users can find the expected format for whatever model they want to use.